lib/tasks/reset.rake

```diff
@@ -0,0 +1,30 @@
+namespace :hyrax do
+  namespace :reset do
+    desc 'Reset fedora / solr and corrisponding database tables w/o clearing other active record tables like users'
+    task works_and_collections: [:environment] do
+      confirm('You are about to delete all works and collections, this is not reversable!')
+      require 'active_fedora/cleaner'
+      ActiveFedora::Cleaner.clean!
+      Hyrax::PermissionTemplateAccess.delete_all
+      Hyrax::PermissionTemplate.delete_all
+      Delayed::Job.delete_all
+
+      # we need to wait till Fedora is done with its cleanup
+      # otherwise creating the admin set will fail
+      while AdminSet.exists?(AdminSet::DEFAULT_ID)
```

Comment:
ActiveFedora::Cleaner is synchronous.  There shouldn't be a need for this.

Reply:
@jcoyne we haven't experienced it to be so. The prevailing theory was that it was async because we've seen multiple instances of the admin set creation failing after calling the cleaner. If you, for example, import 100 works, then run this task w/o the while, it will typically fail. If you run the task a second time it will succeed. Are there any other reasons it would be like that?